__tests__/PropTypesDevelopmentReact15.js

```diff
@@ -875,6 +875,89 @@ describe('PropTypesDevelopmentReact15', () => {
     });
   });
 
+  describe('ElementWithType Types', () => {
+    it('should fail for mismatched component.type', () => {
+      typeCheckFail(
+        PropTypes.elementWithType('div'),
+        <span/>,
+        "Warning: Failed prop type: Invalid prop `testProp` of type [span] supplied to `testComponent`, expected one of type `div`."
+      );
+    });
+
+    it('should pass for matched component.type', () => {
+      typeCheckPass(PropTypes.elementWithType('div'), <div />);
+    });
+
+    it('should pass if no component given', () => {
+      typeCheckPass(PropTypes.elementWithType('div'));
+    });
+
+    describe('factory error', () => {
+      it('should fail if using any value thats not a function nor a string for expectedTypes', () => {
```

Comment:
Wouldn’t this throw, synchronously?

Reply:
what do you mean by that?

Reply:
I mean that `PropTypes.elementWithType(x)` should throw, immediately, if `x` is not a string or a function - it shouldn’t have to wait until propType validations to do so.

Reply:
yes, updated factory spec.